export/mlab_export_test.py

```diff
@@ -0,0 +1,492 @@
+
+    self.assertRaises(mlab_export.TimeOptionError,
+                      mlab_export.assert_start_and_end_times, mock_options)
+
+  def testunit_assert_start_and_end_times_WHEN_too_short_RAISES_TimeOptionError(
+      self):
+    mock_options = mock.Mock()
+    mock_options.length = 3600
+    # Start & end should be at least 'length' apart. Subtract 1 to the
+    # difference too short.
+    mock_options.ts_end = 1401401409 + mock_options.length - 1
+    mock_options.ts_start = 1401401409
+
+    self.assertRaises(mlab_export.TimeOptionError,
+                      mlab_export.assert_start_and_end_times, mock_options)
+
+  def testunit_read_metric_map(self):
+    fake_metric_conf = os.path.join(self._testdata_dir, 'sample_metrics.conf')
+    expected_map = {'cpu_cores.gauge': 'cpu.cores',
+                    'uptime.uptime': 'system.uptime'}
+
+    returned_map = mlab_export.read_metric_map(fake_metric_conf)
+
+    self.assertEqual(returned_map, expected_map)
+
+  @mock.patch('mlab_export.logging.error')
+  def testunit_read_metric_map_WHEN_no_file(self, mock_error):
+    fake_metric_conf = os.path.join(self._testdata_dir, 'no_such_metrics.conf')
+
+    self.assertRaises(SystemExit, mlab_export.read_metric_map, fake_metric_conf)
+    self.assertTrue(mock_error.called)
+
+  @mock.patch('mlab_export.logging.error')
+  def testunit_read_metric_map_WHEN_config_parse_error(self, mock_error):
+    fake_metric_conf = os.path.join(
+        self._testdata_dir, 'sample_metrics_with_error.conf')
+
+    self.assertRaises(SystemExit, mlab_export.read_metric_map, fake_metric_conf)
+    self.assertTrue(mock_error.called)
+
+  @mock.patch('mlab_export.logging.error')
+  def testunit_read_metric_map_WHEN_config_is_missing_metrics_section(
+      self, mock_error):
+    fake_metric_conf = os.path.join(
+        self._testdata_dir, 'sample_metrics_missing_section.conf')
+
+    self.assertRaises(SystemExit, mlab_export.read_metric_map, fake_metric_conf)
+    self.assertTrue(mock_error.called)
+
+  @mock.patch('mlab_export.logging.info')
+  def testunit_get_canonical_names(self, mock_logging):
+    mock_options = enable_show_options(mock.Mock())
+    mock_options.rrddir_prefix = '/var/lib/collectd/rrd/'
+    fake_filename = '/var/lib/collectd/rrd/mlab2.nuq0t/disk-dm-0/disk_time.rrd'
+    value_name = 'write'
+    expected_host = 'mlab2.nuq0t'
+    expected_experiment = 'system'
+    expected_metric = 'disk.swap.io.time.write'
+    mlab_export.HOSTNAME = 'mlab2.nuq0t'
+    mlab_export.METRIC_MAP = {
+        'disk-dm-0.disk_time.write': 'disk.swap.io.time.write'}
+
+    (returned_host, returned_experiment, returned_metric) = (
+        mlab_export.get_canonical_names(
+            fake_filename, value_name, mock_options))
+    
+    self.assertEqual(returned_host, expected_host)
+    self.assertEqual(returned_experiment, expected_experiment)
+    self.assertEqual(returned_metric, expected_metric)
+    self.assertTrue(mock_logging.called)
```

Comment:
Same note about logging not part of the function contract. I'll stop commenting further instances of this.